app/src/main/java/ch/epfl/sweng/rps/models/RandomPlayer.kt

```diff
@@ -0,0 +1,16 @@
+package ch.epfl.sweng.rps.models
+
+import kotlin.random.Random
+
+class RandomPlayer(numberOfPossibleMoves: Int) :
+    ComputerPlayer("randomPlayer", numberOfPossibleMoves) {
+    override fun makeMove(): Hand {
+        val upperRange: Int = numberOfPossibleMoves - 1
+        return when (Random.nextInt(0, upperRange)) {
+            0 -> Hand.ROCK
+            1 -> Hand.PAPER
+            2 -> Hand.SCISSORS
+            else -> throw IllegalStateException("Given number is not correct")
+        }
+    }
+}
```

Comment:
For different types of games e.g. rock paper scissors lizard spock, where the number of possible moves will be 5

Reply:
That doesn't really make sense to me. If you want to support different hands in the future, I would rather provide the set of allowed hands. 

```kotlin
class RandomPlayer(allowedHands: List<Hand>) :
    ComputerPlayer("randomPlayer", allowedHands) {
    override fun makeMove(): Hand {
        val upperRange: Int = allowedHands.size
        return allowedHands[Random.nextInt(0, upperRange)]
    }
}
```

**Also, not that `Random.nextInt(0, x)` generates a number from 0 to x (EXCLUSIVE). So you shouldn't do `val upperRange: Int = numberOfPossibleMoves - 1`.**

You also don't need to provide 0 as a parameter, as by default `Random.nextInt(x)` generates from 0 until x (EXCLUSIVE).

Reply:
I could do that though the hand is not abstracted, and I need to make the change on the level of ComputerPlaye. I could make an abstract class Choice (which would be empty) and follow the way you described.
Thanks for the nextInt

Reply:
I would say that for now, you should drop the option for having a custom number of moves. We will see how we implement it in the future when we support more. But for now it can be confusing for other people, and doesn't serve any purpose yet.